Essentials/Commands/EntityModule.cs

```diff
@@ -251,51 +263,75 @@ public void PowerOn(string name)
         }
 
         [Command("eject", "Ejects a specific player from any block they are seated in, or all players in the server if run with 'all'")]
-        public void Eject(string player)
-        {
-            if (player.ToLower() == "all")
+        public void Eject(string playerName) {
+
+            if (playerName.ToLower() == "all") 
             {
-                foreach (var ap in MySession.Static.Players.GetOnlinePlayers())
-                {
-                    var parent = ap.Character?.Parent;
-                    if (parent == null)
-                        continue;
+                EjectAllPlayers();
+            }
+            else 
+            {
+                EjectSinglePlayer(playerName);
+            }
+        }
+
+        private void EjectAllPlayers() {
 
-                    if (parent is MyShipController c)
+            int ejectedPlayersCount = 0;
+
+            foreach (var grid in MyEntities.GetEntities().OfType<MyCubeGrid>().ToList()) 
+            {
+                foreach (var controller in grid.GetFatBlocks<MyShipController>()) 
+                {
+                    if (controller.Pilot != null) 
                     {
-                        c.RemoveUsers(false);
-                        //Context.Respond($"Ejected {parent.DisplayName} from seat.");
-                        continue;
+                        controller.Use();
```

Comment:
I have not tested if RemoveUsers works fine or not. Use seems to do the job very well for what I have Tested in my previous commit. So I did not see a reason to revert it back to RemoveUsers.